packager/media/formats/mp2t/ts_segmenter_unittest.cc

```diff
@@ -268,6 +260,7 @@ TEST_F(TsSegmenterTest, PassedSegmentDuration) {
   EXPECT_OK(segmenter.Initialize(*stream_info));
   segmenter.InjectTsWriterForTesting(std::move(mock_ts_writer_));
   EXPECT_OK(segmenter.AddSample(*sample1));
+  segmenter.SetSegmentStartedForTesting(false);
```

Comment:
What is the reason for `SetSegmentStartedForTesting`? To skip the file operation?

A better way to do it is to use a working file path. I suggest using a memory filesystem. You can do that by changing `segment template` to:

```c++
options.segment_template = "memory://file$Number$.ts";
```

Reply:
As we are using a mock_ts_writer_ (which was previously writing to file using WriteFile), the buffer is empty. So we get a DCHECK failure at line ts_segmenter.cc:191 (`RETURN_IF_ERROR(segment_buffer_.WriteToFile(segment_file.get()));`). 
By calling `segmenter.SetSegmentStartedForTesting(false)`, we are returning from line ts_segmenter.cc:178 (` if (!segment_started_)  return Status::OK;` ) and line 201 (`listener_->OnNewSegment...`) is not executed.

One way is to create a writeToFile function in ts_writer and mock it. Then I could add the expectation  for muxer_listener::OnNewSegment. 
Let me know what you think.

Reply:
Ah, I see. Another way to address it is to update ```AddPesPacket``` override in line 93:

```c++
  // Similar to the hack above but takes a std::unique_ptr.
  MOCK_METHOD1(AddPesPacketMock, bool(PesPacket* pes_packet));
  bool AddPesPacket(std::unique_ptr<PesPacket> pes_packet,
		  BufferWriter* buffer_writer) override {
    // Push dummy data to |buffer_writer| to keep WriteToFile happy.
    buffer_writer->AppendString("dummy PES packet data");
    // No need to keep the pes packet around for the current tests.
    return AddPesPacketMock(pes_packet.get());
  }
```

Reply:
Have made the changes as per your above comment and updated the pr. Please take a look and let me know.